lib/profile-onboarding.dart

```diff
@@ -0,0 +1,207 @@
+import 'package:flutter/material.dart';
+import 'package:date_time_picker/date_time_picker.dart';
+import 'package:material_segmented_control/material_segmented_control.dart';
+import 'package:project_eureka_flutter/blank_page.dart';
+
+class ProfileOnboarding extends StatefulWidget {
+  @override
+  _ProfileOnboardingState createState() => _ProfileOnboardingState();
+}
+
+class _ProfileOnboardingState extends State<ProfileOnboarding> {
+  bool _profilePage = false;
+  final GlobalKey<FormState> _formKey = GlobalKey<FormState>();
+
+  int _role = 0;
+  String _firstName;
+  String _lastName;
+  String _city;
+  String _birthDate;
+
+  TextStyle appBarTextStyle = TextStyle(
+    color: Colors.black,
+    fontWeight: FontWeight.bold,
+  );
+
+  Map<int, Widget> _children = {
+    0: Container(
+      padding: EdgeInsets.fromLTRB(15.0, 0.0, 15.0, 0.0),
+      child: Text(
+        'None',
+        style: TextStyle(fontSize: 20.0),
+      ),
+    ),
+    1: Container(
+      padding: EdgeInsets.fromLTRB(15.0, 0.0, 15.0, 0.0),
+      child: Text(
+        'Student',
+        style: TextStyle(fontSize: 20.0),
+      ),
+    ),
+    2: Container(
+      padding: EdgeInsets.fromLTRB(15.0, 0.0, 15.0, 0.0),
+      child: Text(
+        'Teacher',
+        style: TextStyle(fontSize: 20.0),
+      ),
+    ),
+  };
+
+  Column segmentedControl() {
+    return Column(
+      children: [
+        Container(
+          padding: EdgeInsets.fromLTRB(0.0, 0.0, 0.0, 10.0),
+          child: Text('Do you have a preference?'),
+        ),
+        MaterialSegmentedControl(
+          children: _children,
+          selectionIndex: _role,
+          borderColor: Colors.grey,
+          selectedColor: Color(0xFF00ADB5),
+          unselectedColor: Colors.white,
+          borderRadius: 32.0,
+          onSegmentChosen: (index) {
+            setState(() {
+              _role = index;
+            });
+          },
+        ),
+        SizedBox(height: 20.0),
+      ],
+    );
+  }
+
+  Column profileTextFormField(
+    String labelText,
+    TextInputAction textInputAction,
+    String validatorMsg,
+    Function onSaved,
+  ) {
+    return Column(
+      children: <Widget>[
+        TextFormField(
+          textCapitalization: TextCapitalization.sentences,
+          keyboardType: TextInputType.text,
+          textInputAction: textInputAction,
+          validator: (value) {
+            if (value.isEmpty) {
```

Comment:
I would also add restrictions: 
- strings consisting of only spaces
- strings not containing letters
- strings containing numbers

Reply:
While I agree there should be some restrictions, it'll be very hard to find the proper regex for that limit certain people. Some people might be named "de Rune" or "X Æ A-12"